src/logic.js

```diff
@@ -390,17 +390,28 @@ define([
             var seen = {},
                 form = this.form;
             function updatePath(mug, property, paths) {
+                function updateExpression(value) {
+                    var expr = new LogicExpression(value, form.xpath),
+                        orig = expr.getText();
+                    expr.updatePath(paths[0], paths[1]);
+                    var updated = expr.getText();
+                    return updated !== orig ? updated : value;
+                }
+
                 var pkey = mug.ufid + " " + property + " " + paths[0];
                 if (seen.hasOwnProperty(pkey)) {
                     return;
                 }
                 seen[pkey] = null;
-                var expr = new LogicExpression(mug.p[property], form.xpath),
-                    orig = expr.getText();
-                expr.updatePath(paths[0], paths[1]);
-                if (orig !== expr.getText()) {
```

Comment:
This used to always evaluate to false for itext properties because `LogicExpression` cannot parse an `ItextItem` so it just returned the original value.